airflow/pipe_naf_reader_dag.py

```diff
@@ -0,0 +1,127 @@
+from datetime import datetime, timedelta
+import os
+import json
+
+from airflow import DAG
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+
+from airflow_ext.gfw.models import DagFactory
+
+
+PIPELINE = "pipe_naf_reader"
+
+class GoogleCloudStoragePrefixSensor(BaseSensorOperator):
+    """
+    Checks for the existence of a file in Google Cloud Storage.
+    """
+    template_fields = ('bucket', 'prefix')
+    ui_color = '#f0eee4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            bucket,
+            prefix,
+            google_cloud_conn_id='google_cloud_storage_default',
+            delegate_to=None,
+            *args,
+            **kwargs):
+
+        super(GoogleCloudStoragePrefixSensor, self).__init__(*args, **kwargs)
+        self.bucket = bucket
+        self.prefix = prefix
+        self.google_cloud_conn_id = google_cloud_conn_id
+        self.delegate_to = delegate_to
+
+    def poke(self, context):
+        self.log.info('Sensor checks existence of : %s, %s', self.bucket, self.prefix)
+        hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.google_cloud_conn_id,
+            delegate_to=self.delegate_to)
+        return len(hook.list(self.bucket, prefix=self.prefix)) > 0
+
+
+def source_exists_sensor(bucket, prefix, dag):
+    return GoogleCloudStoragePrefixSensor(
+        dag=dag,
+        task_id='source_exists',
+        bucket=bucket,
+        prefix=prefix,
+        poke_interval=10,   # check every 10 seconds for a minute
+        timeout=60,
+        retries=24*7,       # retry once per hour for a week
+        retry_delay=timedelta(minutes=60),
+        retry_exponential_backoff=False
+    )
+
+
+class NAFReaderDagFactory(DagFactory):
+
+    def __init__(self, pipeline=PIPELINE, **kwargs):
+        super(NAFReaderDagFactory, self).__init__(pipeline=pipeline, **kwargs)
+
+    def source_date(self):
+        if self.schedule_interval == '@daily':
+            return '{{ ds }}','{{ ds_nodash }}'
+        else:
+            raise ValueError('Unsupported schedule interval {}'.format(self.schedule_interval))
+
+    def build(self, dag_id):
+        config = self.config
+        source_date = self.source_date()
+        config['date'] = source_date[0]
```

Comment:
@smpiano you should have all of this already in the context by using pipe_tools.